yt_dlp/YoutubeDL.py

```diff
@@ -2490,12 +2508,16 @@ def _calc_headers(self, info_dict):
                     values.append(f'Version={cookie.version}')
             info_dict['cookies'] = '; '.join(values)
 
+        res = merge_headers(self.params['http_headers'], info_dict.get('http_headers') or {})
+        if 'Youtubedl-No-Compression' in res:  # deprecated
+            res.pop('Youtubedl-No-Compression', None)
+            res['Accept-Encoding'] = 'identity'
         if 'X-Forwarded-For' not in res:
             x_forwarded_for_ip = info_dict.get('__x_forwarded_for_ip')
             if x_forwarded_for_ip:
                 res['X-Forwarded-For'] = x_forwarded_for_ip
 
-        return res
+        return self._remove_cookie_header(res)
```

Comment:
I'm noticing this will be a problem for #2861, since it needs to return a `CaseInsensitiveDict` here.
The cookie header filter could be moved up in network rework, like:
```py
        res = CaseInsensitiveDict(
            self.params['http_headers'], self._remove_cookie_header(info_dict.get('http_headers')))
        clean_headers(res)
        if 'X-Forwarded-For' not in res:
            x_forwarded_for_ip = info_dict.get('__x_forwarded_for_ip')
            if x_forwarded_for_ip:
                res['X-Forwarded-For'] = x_forwarded_for_ip

        return res
```

Reply:
Im for fixing things when they come up. Having said that, we can set `http_headers` to a `CaseInsensitiveDict` always, and then rely on case insensitivity and simply do `headers.pop("Cookie")`. We will have to see later though